test/e2e/util/proxy.go

```diff
@@ -0,0 +1,149 @@
+package util
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/openshift/api/config/v1"
+	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
+	"github.com/openshift/hypershift/cmd/cluster/core"
+	awsinfra "github.com/openshift/hypershift/cmd/infra/aws"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const CA_CONFIGMAP_KEY = "ca-bundle.crt"
+
+func CreateProxyInstance(t *testing.T, ctx context.Context, client crclient.Client, opts core.CreateOptions, hostedCluster *hyperv1.HostedCluster) *v1.ProxySpec {
```

Comment:
Any chance we can change the current CLI code to use the cert and so we keep the test pretty much as it's by using `clusterOpts.AWSPlatform.EnableProxy = true`?

Reply:
Do you want the CLI to generate the cert as well, or pass it as an argument?
Enabling the proxy cert in CLI by default will break `Public` Clusters created with a proxy, are we fine by that?